File: 0041-first-missing-positive/0041-first-missing-positive.cpp
#include <set>
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        set<int> s;
        for(int x : nums){
            s.insert(x);
        }
        int i = 1;
        while(true){
            if(!s.count(i)) return i;
            i++;
        }
    }
};

File: 0041-first-missing-positive/0041-first-missing-positive.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var firstMissingPositive = function(nums) {
        let s=new Set(nums)
        let i=1
        let found
        while(true){
            if(!s.has(i)) return i
            i++
        }
};